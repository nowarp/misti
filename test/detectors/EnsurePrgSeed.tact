contract C {
    fun test(): Int {
        // Bad: Calling nativeRandom without PRG init
        return nativeRandom();
    }
}
