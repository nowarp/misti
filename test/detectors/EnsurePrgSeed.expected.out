[MEDIUM] EnsurePrgSeed: PRG seed should be initialized before using nativeRandom
test/detectors/EnsurePrgSeed.tact:4:16:
  3 |         // Bad: Calling nativeRandom without PRG init
> 4 |         return nativeRandom();
                     ^
  5 |     }
Help: Use `nativePrepareRandom`, `nativeRandomize`, `nativeRandomizeLt` to initialize the PRG seed or choose the safer `randomInt` function
See: https://nowarp.io/tools/misti/docs/detectors/EnsurePrgSeed